### DISCUSSION TITLE
Running an agent, how do I filter out some of the events

### DISCUSSION THREAD

**Discussion by gustaff-weldon** [open-telemetry/opentelemetry-java-instrumentation#5680] [Q&A] [answered]
We are running the service with agent via 
```
ava -javaagent:path/to/opentelemetry-javaagent.jar \
     -jar myapp.jar
```
We did not exclude any instrumentations from the list (https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/suppressing-instrumentation.md)

We would like to exclude reporting certain http calls made to the service, eg. we have a health `/actuator/health` which generates a lot of events eg.

<img width="1543" alt="Screenshot 2022-03-24 at 09 43 29" src="https://user-images.githubusercontent.com/411756/159876999-3eea0e0b-97c4-473f-8d86-688dd25b05bf.png">

I could not find any configuration for the Tomcat instrumentation (I believe that's the one that reports the events) in https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/tomcat

How can I drop the unnecessary reporting?

I'm looking for sth similar to what JS HTTP instrumentation offers:
```
      new HttpInstrumentation({
        ignoreIncomingPaths: [/\.well-known\/apollo\/server-health/],
      }),
```

How can I achieve that?

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @gustaff-weldon,
You could try configuring a custom sampler using a javaagent [extension](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension), e.g. a [`RuleBasedRoutingSampler`](https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/samplers) from the contrib repo.
Unfortunately the samplers project doesn't have documentation other than Javadocs, but perhaps something like this would suffice:
```java
RuleBasedRoutingSampler.builder(SERVER, Sampler.parentBased(Sampler.alwaysOff))
  .drop(SemanticAttributes.HTTP_TARGET, "/actuator")
  .build()
```
